[libpas] Enable libpas on GTK / WPE

Constellation opened the pull request:
#### fd0426d0c812116d14e6c6b66f6ad07cd1d7b00e
<pre>
[libpas] Enable libpas on GTK / WPE
<a href="https://bugs.webkit.org/show_bug.cgi?id=243201">https://bugs.webkit.org/show_bug.cgi?id=243201</a>

Reviewed by Michael Catanzaro.

This patch enables libpas on GTK and WPE Linux ports if they are X86_64 or ARM64.
This offers 1% improvement in JetStream2 (182.483 v.s. 184.102).
In RAMification score, we first observed regression, but it turned out that this
is because Linux RAMification is not correctly measuring footprint: it is using
RSS instead of private dirty memory, so it includes text region etc. And after using
smaps based footprint calculation, the result is roughly the same between libpas and
bmalloc. Also, we know that text size in WebKit / WebCore singificantly decreases
While text size in JavaScriptCore increases with libpas (because of isoheap implementation
improvement), we can say libpas enablement can be win in performance and memory.

* Source/bmalloc/bmalloc/BPlatform.h:
* Tools/WebKitTestRunner/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/252858@main">https://commits.webkit.org/252858@main</a>
</pre>

mcatanzaro reviewed (approved):
Looks like WPE build is failing:

```
[2023/2197] Linking CXX shared library lib/libTestRunnerInjectedBundle.so
FAILED: lib/libTestRunnerInjectedBundle.so 
ld.lld: error: undefined symbol: jit_heap_runtime_config
ld.lld: error: undefined symbol: jit_heap_try_allocate
ld.lld: error: undefined symbol: jit_heap_get_size
ld.lld: error: undefined symbol: jit_heap_shrink
ld.lld: error: undefined symbol: jit_heap_add_fresh_memory
ld.lld: error: undefined symbol: jit_heap_deallocate
ld.lld: error: undefined symbol: pas_scavenger_disable_shut_down
ld.lld: error: undefined symbol: pas_heap_lock
ld.lld: error: undefined symbol: pas_log_stream
ld.lld: error: undefined symbol: pas_debug_spectrum_dump
ld.lld: error: undefined symbol: pas_debug_spectrum_reset
ld.lld: error: undefined symbol: pas_lock_lock_slow
```

Constellation committed:
[libpas] Enable libpas on GTK / WPE
https://bugs.webkit.org/show_bug.cgi?id=243201

Reviewed by Michael Catanzaro.

This patch enables libpas on GTK and WPE Linux ports if they are X86_64 or ARM64.
This offers 1% improvement in JetStream2 (182.483 v.s. 184.102).
In RAMification score, we first observed regression, but it turned out that this
is because Linux RAMification is not correctly measuring footprint: it is using
RSS instead of private dirty memory, so it includes text region etc. And after using
smaps based footprint calculation, the result is roughly the same between libpas and
bmalloc. Also, we know that text size in WebKit / WebCore singificantly decreases
While text size in JavaScriptCore increases with libpas (because of isoheap implementation
improvement), we can say libpas enablement can be win in performance and memory.

* Source/bmalloc/bmalloc/BPlatform.h:
* Tools/WebKitTestRunner/CMakeLists.txt:

Canonical link: https://commits.webkit.org/252858@main
